### PR TITLE
[ETCM-503] testing in CI takes a lot of time

### PR DESCRIPTION
**Pull request by dmitry-worker**
# Description

FastSyncSpec is failing one test repeatedly for some reason at CI... couldn't reproduce it locally..
Maybe it has something to do with a number of builds running simultaneously. They probably share the same hardware for testing.

# Proposed Solution

Increase the timeout for a test.

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/network/PeerStatisticsSpec.scala [hunk 1]
```diff
@@ -48,7 +48,9 @@ class PeerStatisticsSpec
 
     val statA = stats.stats(alice)
     statA.responsesReceived shouldBe 2
-    statA.lastSeenTimeMillis shouldBe >(statA.firstSeenTimeMillis)
+    // @dv: the corner case when responses are received simultaneously
+    // can ruin the tests for no reason
+    statA.lastSeenTimeMillis shouldBe >=(statA.firstSeenTimeMillis)
```
Comment (aakoshh) [inline src/test/scala/io/iohk/ethereum/network/PeerStatisticsSpec.scala]: Sorry for that. There may be better ways to test this via mocked clocks but I wanted to make sure that the first and last are treated separately. If they were always equal that would be an error.

File: src/test/scala/io/iohk/ethereum/blockchain/sync/FastSyncSpec.scala [hunk 2]
```diff
@@ -22,7 +22,7 @@ class FastSyncSpec
     with FreeSpecBase
     with SpecFixtures
     with WithActorSystemShutDown { self =>
-  implicit val timeout: Timeout = Timeout(10.seconds)
+  implicit val timeout: Timeout = Timeout(30.seconds)
```
Comment (enriquerodbe) [inline src/test/scala/io/iohk/ethereum/blockchain/sync/FastSyncSpec.scala:25]: Does this mean that we have some tests that take more than 10 seconds to complete? We should improve those. If that's out of the scope of this PR maybe we can create a task to improve those tests.

**Review by enriquerodbe — APPROVED** [review]

File: src/test/scala/io/iohk/ethereum/blockchain/sync/FastSyncSpec.scala (hunk 2 above)
Comment (dmitry-worker) [reply src/test/scala/io/iohk/ethereum/blockchain/sync/FastSyncSpec.scala:25]: Locally it runs for 2.6 seconds approximately. Maybe if I'll catch up with the logic of Fast/Regular sync it would be much easier to refactor these tests.

File: src/test/scala/io/iohk/ethereum/network/PeerStatisticsSpec.scala (hunk 1 above)
Comment (dmitry-worker) [reply src/test/scala/io/iohk/ethereum/network/PeerStatisticsSpec.scala]: I have provided a clock instance, please take a look into it, the pull requests has grown bigger though :(

File: src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala [hunk 3]
```diff
@@ -13,8 +13,10 @@ class TimeSlotStats[K, V: Monoid] private (
     // The last written position in the buffer.
     val lastIdx: Int,
     // Ring buffer of slots statistics.
-    val buffer: TimeSlotStats.Buffer[K, V]
-)(implicit clock: Clock) {
+    val buffer: TimeSlotStats.Buffer[K, V],
+    // The time counter counter
+    clock: Clock
+) {
```
Comment (aakoshh) [inline src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala]: What was the problem with the way it was? I used an implicit just to make it a bit easier to return new instances of this otherwise immutable class.

File: src/test/scala/io/iohk/ethereum/utils/MockClock.scala [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+                 zoneId: ZoneId = ZoneId.of("UTC")
+               ) extends Clock {
+  def windByMillis(by: Long): Unit =
+    currentTimeMillis = currentTimeMillis + by
+
+  override def instant(): Instant = Instant.ofEpochMilli(currentTimeMillis)
+  // The following are implemented for completness' sake but not used:
+  override def getZone(): ZoneId = zoneId
+  override def withZone(x: ZoneId): Clock = new MockClock(currentTimeMillis, zoneId)
+}
```
Comment (aakoshh) [inline src/test/scala/io/iohk/ethereum/utils/MockClock.scala]: You may want to configure your editor to add a newline at the end of the file.

**Review by aakoshh — APPROVED** [review]
Thanks for the update, looks good!

As we discussed privately another option would have been to use `Instant` in the `PeerStat` type for `firstSeenTime` and `lastSeenTime` which has nanosecond precision, but this approach ultimately gives you full control.

File: src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala (hunk 3 above)
Comment (aakoshh) [reply src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala]: It also seemed to make some sense that different parts of the app shouldn't have different clocks, so passing around an implicit one to whoever needs it might be better than threading it through. But ultimately it was only passed on in once place in this class so I get it if you don't like it.

Comment (dmitry-worker) [reply src/main/scala/io/iohk/ethereum/network/TimeSlotStats.scala]: Made clock implicit again

File: src/test/scala/io/iohk/ethereum/utils/MockClock.scala (hunk 4 above)
Comment (dmitry-worker) [reply src/test/scala/io/iohk/ethereum/utils/MockClock.scala]: Added newline
